phpBB/phpbb/ban/type/base.php

```diff
@@ -0,0 +1,241 @@
+<?php
+/**
+ *
+ * This file is part of the phpBB Forum Software package.
+ *
+ * @copyright (c) phpBB Limited <https://www.phpbb.com>
+ * @license GNU General Public License, version 2 (GPL-2.0)
+ *
+ * For full copyright and license information, please see
+ * the docs/CREDITS.txt file.
+ *
+ */
+
+namespace phpbb\ban\type;
+
+use phpbb\db\driver\driver_interface;
+
+abstract class base implements type_interface
+{
+	/** @var driver_interface */
+	protected $db;
+
+	/** @var array */
+	protected $excluded;
+
+	/** @var string */
+	protected $bans_table;
+
+	/** @var string */
+	protected $sessions_keys_table;
+
+	/** @var string */
+	protected $sessions_table;
+
+	/** @var \phpbb\user */
+	protected $user;
+
+	/** @var string */
+	protected $users_table;
+
+	/**
+	 * Creates a ban type.
+	 *
+	 * @param driver_interface	$db						A phpBB DBAL object
+	 * @param string			$bans_table				The bans table
+	 * @param string			$users_table			The users table
+	 * @param string			$sessions_table			The sessions table
+	 * @param string			$sessions_keys_table	The sessions keys table
+	 */
+	public function __construct(driver_interface $db, string $bans_table, string $users_table, string $sessions_table, string $sessions_keys_table)
+	{
+		$this->db = $db;
+		$this->bans_table = $bans_table;
+		$this->users_table = $users_table;
+		$this->sessions_table = $sessions_table;
+		$this->sessions_keys_table = $sessions_keys_table;
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public function set_user(\phpbb\user $user): void
+	{
+		// TODO: Implement new logging
```

Comment:
Plan to implement it later?

Reply:
It's included in the ban manager already, removing the todo